library/alloc/src/vec/mod.rs

```diff
@@ -1490,6 +1490,12 @@ impl<T, A: Allocator> Vec<T, A> {
     /// vec.insert(4, 5);
     /// assert_eq!(vec, [1, 4, 2, 3, 5]);
     /// ```
+    ///
+    /// # Time complexity
+    ///
+    /// Takes *O*([`Vec::len`]) time. All items after the insertion index must be
```

Comment:
```suggestion
    /// Takes at worst *O*(*n*) time. All items after the insertion index must be
```
And perhaps specify that if *index* = *n* then it's O(1).

Reply:
I think it's fine -- Big O notation doesn't have to be precise, and this does represent the average (len/2) number of items that have to move, since constants like 1/2 are usually dropped.